dart/lib/src/client.dart

```diff
@@ -188,48 +86,69 @@ abstract class SentryClient {
-  String toString() => '$SentryClient("$postUri")';
-
-  @protected
-  List<int> bodyEncoder(Map<String, dynamic> data, Map<String, String> headers);
-
-  @protected
-  @mustCallSuper
-  Map<String, String> buildHeaders(String authHeader) {
-    final headers = {
-      'Content-Type': 'application/json',
-    };
-
-    if (authHeader != null) {
-      headers['X-Sentry-Auth'] = authHeader;
+  SentryEvent _processEvent(
+    SentryEvent event, {
+    dynamic hint,
+    List<EventProcessor> eventProcessors,
+  }) {
+    for (final processor in eventProcessors) {
+      try {
+        event = processor(event, hint);
+      } catch (err) {
+        options.logger(
+          SentryLevel.error,
+          'An exception occurred while processing event by a processor : $err',
+        );
+      }
+      if (event == null) {
+        options.logger(SentryLevel.debug, 'Event was dropped by a processor');
+        break;
+      }
     }
-
-    return headers;
+    return event;
   }
-}
 
-/// A response from Sentry.io.
-///
-/// If [isSuccessful] the [eventId] field will contain the ID assigned to the
-/// captured event by the Sentry.io backend. Otherwise, the [error] field will
-/// contain the description of the error.
-@immutable
-class SentryResponse {
-  const SentryResponse.success({@required this.eventId})
-      : isSuccessful = true,
-        error = null;
-
-  const SentryResponse.failure(this.error)
-      : isSuccessful = false,
-        eventId = null;
-
-  /// Whether event was submitted successfully.
-  final bool isSuccessful;
-
-  /// The ID Sentry.io assigned to the submitted event for future reference.
-  final String eventId;
-
-  /// Error message, if the response is not successful.
-  final String error;
+  SentryEvent _applyScope({
+    @required SentryEvent event,
+    @required Scope scope,
+  }) {
+    if (scope != null) {
+      // Merge the scope transaction.
+      if (event.transaction == null) {
+        event = event.copyWith(transaction: scope.transaction);
+      }
+
+      // Merge the user context.
+      if (event.userContext == null) {
+        event = event.copyWith(userContext: scope.user);
+      }
+
+      // Merge the scope fingerprint.
+      if (event.fingerprint == null) {
+        event = event.copyWith(fingerprint: scope.fingerprint);
+      }
+
+      // Merge the scope breadcrumbs.
+      if (event.breadcrumbs == null) {
+        event = event.copyWith(breadcrumbs: scope.breadcrumbs);
+      }
+
+      // TODO add tests
+      // Merge the scope tags.
+      event = event.copyWith(
+          tags: scope.tags.map((key, value) => MapEntry(key, value))
+            ..addAll(event.tags ?? {}));
+
+      // Merge the scope extra.
+      event = event.copyWith(
+          extra: scope.extra.map((key, value) => MapEntry(key, value))
+            ..addAll(event.extra ?? {}));
+
+      // Merge the scope level.
+      if (event.level == null) {
+        event = event.copyWith(level: scope.level);
+      }
```

Comment:
Scope level has precedence here:
https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryClient.java#L327-L330

Reply:
also, missing Contexts, created a card for that https://github.com/getsentry/sentry-dart/projects/1#card-47942441
it doesn't need to be on this PR